src/DynamoPackages/PackageDirectoryBuilder.cs

```diff
@@ -124,44 +124,42 @@ private void WritePackageHeader(Package package, IDirectoryInfo rootDir)
         }
 
 
-        public void CopyFilesIntoPackageDirectory(IEnumerable<string> files, IDirectoryInfo dyfDir,
+        internal void CopyFilesIntoPackageDirectory(IEnumerable<string> files, IDirectoryInfo dyfDir,
                                                           IDirectoryInfo binDir, IDirectoryInfo extraDir)
         {
-            foreach (var file in files)
-            {
-                // If this file is null, then just continue
-                if (file == null)
-                {
-                    continue;
-                }
+            // normalize the paths to ensure correct comparison
+            var dyfDirPath = NormalizePath(dyfDir.FullName);
+            var binDirPath = NormalizePath(binDir.FullName);
+            var extraDirPath = NormalizePath(extraDir.FullName);
 
+            foreach (var file in files.Where(x => x != null))
```

Comment:
Why there is null check here? We should ideally pass only those files which exist. May be I am missing something here, but I think files will not be null here - but placing a null inside an IEnumerable is not good.